### ISSUE TITLE
mingw64\bin\mingw32-make.exe is corrupted

### ISSUE THREAD

**Issue by zdm** [Alexpux/MSYS2-packages#842]
Hi.
After today update I found, that mingw64\bin\mingw32-make.exe was updated to corrupted version. that does nothing. 
It just shows empty output.
Please, fix this issue.

**Comment by mati865** [comment]
Can you provide specific case?

> It just shows empty output.

```
$ mingw32-make
mingw32-make: *** No targets specified and no makefile found.  Stop.

mati865@DESKTOP-0J8VPHJ MINGW64 ~
$ mingw32-make -h
Usage: mingw32-make [options] [target] ...
Options:
  -b, -m                      Ignored for compatibility.
  -B, --always-make           Unconditionally make all targets.
  -C DIRECTORY, --directory=DIRECTORY
                              Change to DIRECTORY before doing anything.
  -d                          Print lots of debugging information.
  --debug[=FLAGS]             Print various types of debugging information.
  -e, --environment-overrides
                              Environment variables override makefiles.
  --eval=STRING               Evaluate STRING as a makefile statement.
  -f FILE, --file=FILE, --makefile=FILE
                              Read FILE as a makefile.
  -h, --help                  Print this message and exit.
  -i, --ignore-errors         Ignore errors from recipes.
  -I DIRECTORY, --include-dir=DIRECTORY
                              Search DIRECTORY for included makefiles.
  -j [N], --jobs[=N]          Allow N jobs at once; infinite jobs with no arg.
  -k, --keep-going            Keep going when some targets can't be made.
  -l [N], --load-average[=N], --max-load[=N]
                              Don't start multiple jobs unless load is below N.
  -L, --check-symlink-times   Use the latest mtime between symlinks and target.
  -n, --just-print, --dry-run, --recon
                              Don't actually run any recipe; just print them.
  -o FILE, --old-file=FILE, --assume-old=FILE
                              Consider FILE to be very old and don't remake it.
  -O[TYPE], --output-sync[=TYPE]
                              Synchronize output of parallel jobs by TYPE.
  -p, --print-data-base       Print make's internal database.
  -q, --question              Run no recipe; exit status says if up to date.
  -r, --no-builtin-rules      Disable the built-in implicit rules.
  -R, --no-builtin-variables  Disable the built-in variable settings.
  -s, --silent, --quiet       Don't echo recipes.
  -S, --no-keep-going, --stop
                              Turns off -k.
  -t, --touch                 Touch targets instead of remaking them.
  --trace                     Print tracing information.
  -v, --version               Print the version number of make and exit.
  -w, --print-directory       Print the current directory.
  --no-print-directory        Turn off -w, even if it was turned on implicitly.
  -W FILE, --what-if=FILE, --new-file=FILE, --assume-new=FILE
                              Consider FILE to be infinitely new.
  --warn-undefined-variables  Warn when an undefined variable is referenced.

This program built for x86_64-w64-mingw32
Report bugs to <bug-make@gnu.org>
```

Also tested with random code https://paste.ubuntu.com/24165635/

**Comment by zdm** [comment]
hi,
as you can see, it worked before update and stop working after.

```
Microsoft Windows [Version 10.0.14393]
(c) 2016 Microsoft Corporation. All rights reserved.

d:\devel\msys2\mingw64\bin>.\mingw32-make.exe
mingw32-make: *** No targets specified and no makefile found.  Stop.











d:\devel\msys2\mingw64\bin>update-msys2
:: Synchronizing package databases...
 mingw32                                                              319.8 KiB   371K/s 00:01 [#######################################################] 100%
 mingw32.sig                                                           96.0   B  93.8K/s 00:00 [#######################################################] 100%
 mingw64                                                              319.8 KiB  4.00M/s 00:00 [#######################################################] 100%
 mingw64.sig                                                           96.0   B  0.00B/s 00:00 [#######################################################] 100%
 msys is up to date                                                     0.0   B  0.00B/s 00:00 [-------------------------------------------------------]   0%
:: Starting core system upgrade...
 there is nothing to do
:: Starting full system upgrade...
resolving dependencies...
looking for conflicting packages...

Packages (7) mingw-w64-x86_64-binutils-2.28-1  mingw-w64-x86_64-crt-git-5.0.0.4795.e3d96cb1-1  mingw-w64-x86_64-gcc-6.3.0-2
             mingw-w64-x86_64-gcc-libs-6.3.0-2  mingw-w64-x86_64-headers-git-5.0.0.4797.31e66d7e-1  mingw-w64-x86_64-libiconv-1.15-1
             mingw-w64-x86_64-make-4.2.1-1

Total Download Size:    49.27 MiB
Total Installed Size:  350.38 MiB
Net Upgrade Size:        0.85 MiB

:: Proceed with installation? [Y/n]
:: Retrieving packages...
 mingw-w64-x86_64-libiconv-1.15-1-any                                 613.1 KiB   509K/s 00:01 [#######################################################] 100%
 mingw-w64-x86_64-gcc-libs-6.3.0-2-any                                562.4 KiB  1484K/s 00:00 [#######################################################] 100%
 mingw-w64-x86_64-binutils-2.28-1-any                                  12.9 MiB  3.25M/s 00:04 [#######################################################] 100%
 mingw-w64-x86_64-headers-git-5.0.0.4797.31e66d7e-1-any                 5.0 MiB  2.50M/s 00:02 [#######################################################] 100%
 mingw-w64-x86_64-crt-git-5.0.0.4795.e3d96cb1-1-any                     2.9 MiB  2.24M/s 00:01 [#######################################################] 100%
 mingw-w64-x86_64-gcc-6.3.0-2-any                                      27.2 MiB  2.63M/s 00:10 [#######################################################] 100%
 mingw-w64-x86_64-make-4.2.1-1-any                                    133.7 KiB  4.84M/s 00:00 [#######################################################] 100%
(7/7) checking keys in keyring                                                                 [#######################################################] 100%
(7/7) checking package integrity                                                               [#######################################################] 100%
(7/7) loading package files                                                                    [#######################################################] 100%
(7/7) checking for file conflicts                                                              [#######################################################] 100%
(7/7) checking available disk space                                                            [#######################################################] 100%
:: Processing package changes...
(1/7) upgrading mingw-w64-x86_64-libiconv                                                      [#######################################################] 100%
(2/7) upgrading mingw-w64-x86_64-gcc-libs                                                      [#######################################################] 100%
(3/7) upgrading mingw-w64-x86_64-binutils                                                      [#######################################################] 100%
(4/7) upgrading mingw-w64-x86_64-headers-git                                                   [#######################################################] 100%
(5/7) upgrading mingw-w64-x86_64-crt-git                                                       [#######################################################] 100%
(6/7) upgrading mingw-w64-x86_64-gcc                                                           [#######################################################] 100%
(7/7) upgrading mingw-w64-x86_64-make                                                          [#######################################################] 100%
warning: bash-4.4.012-1 is up to date -- skipping
warning: pacman-5.0.1-2 is up to date -- skipping
warning: pacman-mirrors-20160112-1 is up to date -- skipping
warning: msys2-runtime-2.7.0-1 is up to date -- skipping
warning: diffutils-3.5-1 is up to date -- skipping
warning: man-db-2.7.4-1 is up to date -- skipping
warning: tar-1.29-1 is up to date -- skipping
warning: wget-1.19.1-1 is up to date -- skipping
warning: ctags-5.8-1 is up to date -- skipping
warning: whois-5.2.9-1 is up to date -- skipping
warning: patch-2.7.5-1 is up to date -- skipping
 there is nothing to do
warning: mingw-w64-x86_64-gcc-6.3.0-2 is up to date -- skipping
warning: mingw-w64-x86_64-make-4.2.1-1 is up to date -- skipping
 there is nothing to do
error: no targets specified (use -h for help)

Cache directory: /var/cache/pacman/pkg/
:: Do you want to remove ALL files from cache? [y/N] y
removing all files from cache...

Database directory: /var/lib/pacman/
:: Do you want to remove unused repositories? [Y/n] y
removing unused sync repositories...




d:\devel\msys2\mingw64\bin>.\mingw32-make.exe

d:\devel\msys2\mingw64\bin>.\mingw32-make.exe -h

d:\devel\msys2\mingw64\bin>
```

**Comment by mati865** [comment]
There must be some dependency missing:
```
$ pacman -Syu
:: Synchronizing package databases...
 mingw32 is up to date
 mingw64 is up to date
 msys is up to date
:: Starting core system upgrade...
 there is nothing to do
:: Starting full system upgrade...
 there is nothing to do

$ /mingw64/bin/mingw32-make --version
GNU Make 4.2.1
Built for x86_64-w64-mingw32
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

**Comment by zdm** [comment]
maybe package configured improperly, if dependency is not installing?

**Comment by mati865** [comment]
When you have hundreds of installed packages it's hard to find all dependencies.
You can start with installing mingw versions of `gettext` and `readline`

**Comment by TBBle** [comment]
Based on #828, `mingw-w64-x86_64-gettext` seems a likely culprit.

**Comment by zilti** [comment]
Maybe the PKGBUILD should get corrected then. The issue is still there

**Comment by TBBle** [comment]
Suggest closing in favour of #828: It currently reports a failure rather than doing nothing.
```
$ mingw32-make.exe --version
C:/msys64/mingw64/bin/mingw32-make.exe: error while loading shared libraries: libintl-8.dll: cannot open shared object file: No such file or directory
```

**Comment by lazka** [comment]
Closing for #828
